### PR TITLE
[FEAT]: Backend Endpoint for Sending Verification Token

### PR DESCRIPTION
**Pull request by Ofektom**
### Description
Implement a backend endpoint that sends a verification token to the user’s registered email upon account creation or email update. This token will be used to verify the user’s email address.

### Changes made
- Created POST endpoint to send verification token to user email
- Created methods for business logic for the endpoint to be implemented
- Created POST endpoint to accept the 6 digit token

#### Endpoint
`/api/v1/auth/request/token`

#### Request Body
```json
{
    "email": "okpohot@gmail.com"
}
```

#### Response

"Verification email sent successfully"

#### Related Issue
This feature is related to the [issue](https://github.com/hngprojects/hng_boilerplate_java_web/issues/629)

### CODE REVIEW

**Comment by Ofektom** [comment]
@Am0du @Adewale66 please review

File: src/main/java/hng_java_boilerplate/user/serviceImpl/UserServiceImpl.java [hunk 1]
```diff
@@ -158,6 +172,40 @@ private void createPasswordResetTokenForUser(User user, String token) {
         passwordResetTokenRepository.save(newlyCreatedPasswordResetToken);
     }
 
+    @Override
+    public ResponseEntity<String> resetPassword(String token, ResetPasswordDto passwordDto) {
```
Comment (Adewale66) [inline src/main/java/hng_java_boilerplate/user/serviceImpl/UserServiceImpl.java]: This won't return a JSON back, it'll return a string

**Review by Adewale66 — CHANGES_REQUESTED** [review]

File: src/main/java/hng_java_boilerplate/user/serviceImpl/UserServiceImpl.java (hunk 1 above)
Comment (Ofektom) [reply src/main/java/hng_java_boilerplate/user/serviceImpl/UserServiceImpl.java]: I don't think mail sending endpoint return JSON, it just returns a string successful mail sent

**Comment by Ofektom** [comment]
@Adewale66 @Am0du confirm this

**Review by Adewale66 — DISMISSED** [review]
LGTM

**Comment by Am0du** [comment]
You have conflict @Ofektom

**Comment by Ofektom** [comment]
@lurldgbodex @gurumarv @Fifetoyi please review and merge

**Comment by Ofektom** [comment]
@Am0du Conflict resolved

**Comment by Fifetoyi** [comment]
some conflicts here

**Comment by Ofektom** [comment]
@Fifetoyi resolved

**Review by Fifetoyi — APPROVED** [review]
lgtm

**Pull request merged**